redhat-appstudio/service-provider-integration-operator
# fix SVPI-232

### What does this PR do?
Allowing not to put secret.type or put "Opaque".
In this case SPIAccessTokenBinding will create Opaque secret with "token" field filled with token data 

### Screenshot/screencast of this PR
n/a

### What issues does this PR fix or reference?
https://issues.redhat.com/projects/SVPI/issues/SVPI-232

###  How to test this PR ?

- Create a SPIAccessTokenBinding with spec.secret.type : Opaque OR without this field
- Make sure it's status.phase == Injected (may require uploading Access Token with OAuth or manually)
- Check the Secret by name mentioned in status.syncedObjectRef.name. It should have "token" key with non empty value (token data)
